.github/workflows/repackage.yml

```diff
@@ -0,0 +1,203 @@
+            fi
+          done
+          touch "${BUILD_DATA_DIR}/cache/cache.yml"
+
+      - name: Restore manifest
+        uses: actions/download-artifact@v3
+        with:
+          # this is archived by the crawl workflow
+          # see .github/workflows/crawl.yml for details
+          name: kernel-crawler-manifest
+
+      - name: List packages
+        run: make list-files
+
+      - name: Split package file
+        id: split-packages
+        run: |
+          import math
+          import os
+          import json
+
+          def chunks(data, chunk_size):
+            c = []
+            for i in range(chunk_size):
+              c.append(data[i::chunk_size])
+            return c
+
+          packages = []
+          with open('.build-data/packages.txt') as pkg:
+            packages = list(map(str.strip, pkg.readlines()))
+
+          parallel_jobs = math.ceil(len(packages) / 500)
+          if parallel_jobs > 32:
+            parallel_jobs = 32
+
+          jobs = chunks(packages, parallel_jobs)
+          parallel = list(range(parallel_jobs))
+
+          combined = {id: jobs[id] for id in range(parallel_jobs)}
+
+          with open(os.environ['GITHUB_OUTPUT'], 'w') as output:
+            output.write(f'parallel-jobs-number={parallel_jobs}\n')
+            output.write(f'parallel-array={json.dumps(parallel)}\n')
+
+          with open('jobs.json', 'w') as j:
+            j.write(json.dumps(combined))
+
+        shell: python
+
+      # This is primarily for sanity checking
+      - name: Archive package list
+        uses: actions/upload-artifact@v3
+        with:
+          name: kernel-repackage-list
+          path: |
+            .build-data/packages.txt
+
+      - name: Archive Jobs and Cache
+        uses: actions/upload-artifact@v3
+        with:
+          name: package-jobs
+          path: |
+            .build-data/cache/cache.yml
+            jobs.json
+
+  repackage:
+    runs-on: ubuntu-latest
+    needs: split-tasks
+    if: ${{ needs.split-tasks.outputs.parallel-jobs-number > 0 }}
+    strategy:
+      fail-fast: false
+      matrix:
+        packer: ${{ fromJSON(needs.split-tasks.outputs.parallel-array) }}
+    env:
+      MANIFEST_FILE: ./kernel-package-lists/manifest.yml
+      BUILD_DATA_DIR: .build-data
+    steps:
+      - uses: actions/checkout@v3
+      - uses: google-github-actions/auth@v1
+        with:
+          credentials_json: '${{ secrets.GOOGLE_CREDENTIALS_KERNEL_CACHE }}'
+
+      - uses: google-github-actions/setup-gcloud@v1
+
+      - uses: ./.github/actions/env
+
+      - name: Build packers
+        run: make packers
+
+      - name: Restore manifest
+        uses: actions/download-artifact@v3
+        with:
+          # this is archived by the crawl workflow
+          # see .github/workflows/crawl.yml for details
+          name: kernel-crawler-manifest
+
+      - name: Download Jobs and Cache
+        uses: actions/download-artifact@v3
+        with:
+          name: package-jobs
```

Comment:
all the uses expect the files to exist in what is essentially `github.workspace`, so I think this is fine for now at least

Reply:
As long as they are not expected in a subdirectory, yeah, should be fine. I'm just mentioning this because GHA might be picking a file that already exists in the repo instead of the one being downloaded here, which would mean it uses the wrong file without failing.